parsl/data_provider/data_manager.py

```diff
@@ -24,7 +24,7 @@ def __init__(self, dfk):
         self.dfk = dfk
         self.globus = None
 
-    def stage_in(self, file, executor):
+    def stage_in(self, file, executor, parent_fut):
```

Comment:
If this remains optional, I favor `parent_fut=None` to make explicit that it is optional.